### PR TITLE
fix queue size metrics with priorities

### PR DESCRIPTION
**Pull request by syphar**
Looking at queue sizes I saw that these are "hanging" when going to zero: 

grafik

When a certain priority has a queued item at any point in time, it's added to the metric. When we don't update the value any more, the last value is used. With the old logic, that would mean when the queued item count for a certain priority goes to zero after being >0, the old value is kept. 

The first solution is to just reset the metric before inserting all current values. We see how this would look like in the end of the screenshot, where I did a server restart, so the metric content was reset. 

What is not so nice with this is that it leads to gaps in the graph, which also makes using some of the prometheus formular harder. So for the priorities we use the most, I insert `0` as value, if there is nothing queued.

### CODE REVIEW

File: src/metrics/mod.rs [hunk 1]
```diff
@@ -307,7 +307,14 @@ impl ServiceMetrics {
...
         let queue_pending_count = queue.pending_count_by_priority()?;
 
-        for (priority, count) in queue_pending_count.iter() {
+        // we reset the metric so we won't keep old gauge values around
+        // when a certain priority is no longer used in the queue.
+        self.queued_crates_count_by_priority.reset();
+
+        // for commonly used priorities we want the value to be zero, and not missing,
+        // when there are no items in the queue with that priority.
+        for priority in 0..=20 {
```
[the remark this answers is not in the document]
Comment (syphar) [reply src/metrics/mod.rs:316]: that's true, I actually wanted to return _all_ priorities, while only setting to zero for this subset. 

But it seems like when implementing, I forgot this :) 

I'll update the logic.

[the remark this answers is not in the document]
Comment (Nemo157) [reply src/metrics/mod.rs:316]: It seems like that would result in the same issue, other priorities would start hanging at their values, or does the hanging only apply to the summed metric in grafana and as long as at least one value within range is 0 it'll sum to 0?

[the remark this answers is not in the document]
Comment (syphar) [reply src/metrics/mod.rs:316]: The reset call above is the one that prevents the hanging, 

The zeros make formulas nicer

[the remark this answers is not in the document]
Comment (Nemo157) [reply src/metrics/mod.rs:316]: Ah right, it's hanging on the docs.rs side, not grafana :+1:

**Review by Nemo157 — COMMENTED** [review]
LGTM, but despite CI being green the tests didn't actually run cause of #2307

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**